### PR TITLE
20200121 zrq ansible

### PR DESCRIPTION
**Pull request by Zarquan**
Notes and playbooks to deploy Hadoop and Spark using Ansible.
Addresses issues #43, #44 and #45.

### CODE REVIEW

File: experiments/zrq/ansible/16-config-yarn-masters.yml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                <name>yarn.resourcemanager.scheduler.class</name>
+                <value>org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity.CapacityScheduler</value>
+            </property>
+                +-->
+
+            <!--+
+                | Maximum limit of memory to allocate to each container request at the Resource Manager.
+            <property>
+                <name>yarn.scheduler.maximum-allocation-mb</name>
+                <value>15000</value>
```
Comment (stvoutsin) [inline experiments/zrq/ansible/16-config-yarn-masters.yml:133]: Perhaps a future issue, to make the max/min allocation values configurable

File: experiments/zrq/ansible/11-install-hadoop.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+#
+
+#
+# This assumes we need the same binaries installed on the masters and workers.
+#
+# This uses version hadoop-3.2.1, which isn't listed on the Hadoop wiki ?
```
Comment (stvoutsin) [inline experiments/zrq/ansible/11-install-hadoop.yml:26]: Any particular reason for using 3.2.1 ? Is this related to the Spark version you are using?

Comment (Zarquan) [reply experiments/zrq/ansible/11-install-hadoop.yml:26]: Good question - It was the most recent available at the time.

Comment (Zarquan) [reply experiments/zrq/ansible/11-install-hadoop.yml:26]: Since then, there has been a newer release, 3.3, which claims to solve (some of) the issues with Java 11.
https://cwiki.apache.org/confluence/display/HADOOP/Hadoop+3.3+Release

File: experiments/zrq/ansible/16-config-yarn-masters.yml (hunk 1 above)
Comment (Zarquan) [reply experiments/zrq/ansible/16-config-yarn-masters.yml:133]: I went with the defaults to start with (most of these settings are commented out in the initial deploy).
We need to have tests in place to demonstrate why we need to change these values, and be able to re-run the tests to check what happens if we change the settings.

File: experiments/zrq/ansible/11-install-hadoop.yml (hunk 2 above)
Comment (Zarquan) [reply experiments/zrq/ansible/11-install-hadoop.yml:26]: Created a new issue to update the deployment to the newer version (#93).

File: experiments/zrq/ansible/16-config-yarn-masters.yml (hunk 1 above)
Comment (Zarquan) [reply experiments/zrq/ansible/16-config-yarn-masters.yml:133]: Added new issues to document the settings we need for Hadoop(#95) and Spark(#96).
